common/src/main/java/org/wso2/testgrid/common/Script.java

```diff
@@ -72,7 +72,7 @@ public String toString() {
     }
 
     public Properties getScriptParameters() {
-        return scriptParameters;
+        return scriptParameters != null ? scriptParameters : new Properties();
```

Comment:
Shall we initialize the scriptParameters with a Properties instance? Otherwise, each get will create a new Properties instance unnecessarily.